### PR TITLE
Fix/issue 2415 for RadicalPilotExecutor 

### PR DESCRIPTION
**Pull request by AymenFJA**
# Description

This PR:

- Fix issue #2415 and adapt `parsl_resource_specifications` as the only way to set task resources.
- Fix a bug when setting an exception for MPI functions.
- Cleanup

### CODE REVIEW

File: parsl/executors/radical/executor.py [hunk 1]
```diff
@@ -200,22 +204,25 @@ def task_state_cb(self, task, state):
...
+                                else:
+                                    raise ValueError("Unknown exception-like type received: {}".format(type(s)))
+                            except Exception as e:
+                                parsl_task.set_exception(
+                                    DeserializationError("Received exception, but handling also threw an exception: {}".format(e)))
+                        else:
+                            parsl_task.set_exception(eval(task.exception))
                     else:
-                        parsl_task.set_exception('Unknow failure for this task')
+                        parsl_task.set_exception('Task failed for unknow reason')
```
Comment (benclifford) [inline parsl/executors/radical/executor.py:225]: should be `unknown` with an extra N?

File: parsl/executors/radical/executor.py [hunk 2]
```diff
@@ -361,14 +363,25 @@ def unwrap(self, func, args):
...
         """
         Convert parsl function to RADICAL-Pilot rp.TaskDescription
         """
 
         task = rp.TaskDescription()
         task.name = func.__name__
         task.timeout = kwargs.get('walltime', 0.0)
+
+        if parsl_resource_specification and isinstance(parsl_resource_specification, dict):
+            logger.debug('mapping parsl resource sepcfications >> rp resource sepcfications')
```
Comment (benclifford) [inline parsl/executors/radical/executor.py:376]: specifications

File: parsl/executors/radical/executor.py [hunk 3]
```diff
@@ -361,14 +363,25 @@ def unwrap(self, func, args):
...
+            for key, val in parsl_resource_specification.items():
+                if key not in task.as_dict():
+                    key = PARSL_RP_RESOURCE_MAP.get(key, None)
+                    if not key:
+                        logger.warning('ignoring task {0} as it is not supported by RP'.format(key))
```
Comment (benclifford) [inline parsl/executors/radical/executor.py:381]: this is ignoring a resource specification key, not a task?

Comment (AymenFJA) [reply parsl/executors/radical/executor.py:381]: This will only ignore the key.

Comment (benclifford) [reply parsl/executors/radical/executor.py:381]: maybe log message is better: ignoring key 'memory'...

**Pull request merged**